File: app/src/main/java/com/forkmang/activity/LoginActivity.java
package com.forkmang.activity;

import static com.forkmang.helper.Constant.SUCCESS_CODE;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.forkmang.R;
import com.forkmang.helper.Constant;
import com.forkmang.helper.StorePrefrence;
import com.forkmang.helper.Utils;
import com.forkmang.network_call.Api;
import com.google.gson.Gson;
import com.google.gson.JsonObject;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity {

    StorePrefrence storePrefrence;
    Context ctx = LoginActivity.this;
    ProgressBar progress_bar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        Button BtnLogin = findViewById(R.id.BtnLogin);
        Button BtnReg = findViewById(R.id.BtnReg);
        TextView TextGuest = findViewById(R.id.TextGuest);
        progress_bar= findViewById(R.id.progress_bar);
        storePrefrence=new StorePrefrence(ctx);

        TextGuest.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
              if(Utils.isNetworkAvailable(ctx))
              {
                  callapi_loginasguest();
              }
              else{
                  Toast.makeText(ctx,Constant.NETWORKEROORMSG,Toast.LENGTH_SHORT).show();
              }
            }
        });


        BtnReg.setOnClickListener(v -> {
            if(storePrefrence.getString(Constant.NAME).length() == 0)
            {
                final Intent mainIntent = new Intent(LoginActivity.this, RegisterActivity.class);
                startActivity(mainIntent);
                //finish();
            }
            else{
                Toast.makeText(ctx,"User already registered please click login", Toast.LENGTH_LONG).show();
            }
            //finish();
        });


        BtnLogin.setOnClickListener(v -> {
            final Intent mainIntent = new Intent(LoginActivity.this, LoginFormActivity.class);
            startActivity(mainIntent);
            //finish();
        });
    }


    private void callapi_loginasguest()
    {
        //showProgress();
        progress_bar.setVisibility(View.VISIBLE);
        String identifier="identifier123";

        Api.getInfo().login_guest(identifier).
                enqueue(new Callback<JsonObject>() {
                    @Override
                    public void onResponse(Call<JsonObject> call, Response<JsonObject> response) {
                        try{

                            //Log.d("Result", jsonObject.toString());
                            if(response.code() == Constant.SUCCESS_CODE_n)
                            {
                                JSONObject jsonObject = new JSONObject(new Gson().toJson(response.body()));
                                if(jsonObject.getString("status").equalsIgnoreCase(SUCCESS_CODE))
                                {
                                    progress_bar.setVisibility(View.GONE);
                                    storePrefrence.setString(Constant.IDENTFIER, jsonObject.getJSONObject("data").getString("identifier"));
                                    storePrefrence.setString(Constant.TOKEN_LOGIN,"");
                                    storePrefrence.setString("id", jsonObject.getJSONObject("data").getString("id"));

                                    final Intent mainIntent = new Intent(ctx, DashBoard_Activity.class);
                                    startActivity(mainIntent);
                                    finish();

                                }
                                else{
                                    progress_bar.setVisibility(View.GONE);
                                    Toast.makeText(ctx, "Error occur please try again", Toast.LENGTH_LONG).show();
                                }
                            }
                            else if(response.code() == Constant.ERROR_CODE)
                            {
                                JSONObject jsonObject = new JSONObject(response.errorBody().string());
                                if(jsonObject.getString("status").equalsIgnoreCase(String.valueOf(Constant.ERROR_CODE)))
                                {
                                    progress_bar.setVisibility(View.GONE);
                                    String error_msg = jsonObject.getString("message") ;
                                    Toast.makeText(ctx,error_msg,Toast.LENGTH_SHORT).show();

                                }
                                else{
                                    progress_bar.setVisibility(View.GONE);
                                    Toast.makeText(ctx, "Error occur please try again", Toast.LENGTH_LONG).show();
                                }


                            }
                        }
                        catch (JSONException | IOException ex)
                        {
                            ex.printStackTrace();
                            progress_bar.setVisibility(View.GONE);
                            Toast.makeText(ctx, "Error occur please try again", Toast.LENGTH_LONG).show();
                        }
                    }
                    @Override
                    public void onFailure(Call<JsonObject> call, Throwable t) {
                        Toast.makeText(ctx, "Error occur please try again", Toast.LENGTH_LONG).show();
                        //stopProgress();
                        progress_bar.setVisibility(View.GONE);
                    }
                });
    }
}

File: app/src/main/java/com/forkmang/activity/LoginActivity.kt
package com.forkmang.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.forkmang.databinding.ActivityLoginBinding
import com.forkmang.helper.Constant
import com.forkmang.helper.Constant.ERROR_CODE
import com.forkmang.helper.Constant.IDENTFIER
import com.forkmang.helper.Constant.NAME
import com.forkmang.helper.Constant.NETWORKEROORMSG
import com.forkmang.helper.Constant.SUCCESS_CODE
import com.forkmang.helper.Constant.SUCCESS_CODE_n
import com.forkmang.helper.Constant.TOKEN_LOGIN
import com.forkmang.helper.StorePrefrence
import com.forkmang.helper.Utils
import com.forkmang.helper.showToastMessage
import com.forkmang.network_call.Api
import com.google.gson.Gson
import com.google.gson.JsonObject
import org.json.JSONException
import org.json.JSONObject
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.io.IOException

class LoginActivity : AppCompatActivity() {

    private val storePrefrence by lazy { StorePrefrence(this) }
    var ctx: Context = this@LoginActivity
    private val binding by lazy { ActivityLoginBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        binding.TextGuest.setOnClickListener {
            if (Utils.isNetworkAvailable(ctx)) {
                callApiLoginAsGuest()
            } else {
                showToastMessage(NETWORKEROORMSG)
            }
        }
        binding.BtnReg.setOnClickListener {
            if (storePrefrence.getString(
                    NAME
                )?.isEmpty() == true
            ) {
                val mainIntent = Intent(this@LoginActivity, RegisterActivity::class.java)
                startActivity(mainIntent)
                //finish();
            } else {
                showToastMessage("User already registered please click login")
            }
        }
        binding.BtnLogin.setOnClickListener {
            val mainIntent = Intent(this@LoginActivity, LoginFormActivity::class.java)
            startActivity(mainIntent)
        }
    }

    private fun callApiLoginAsGuest() {
        binding.progressBar.visibility = View.VISIBLE
        val identifier = "identifier123"
        Api.info.loginGuest(identifier)?.enqueue(object : Callback<JsonObject?> {
            override fun onResponse(
                call: Call<JsonObject?>,
                response: Response<JsonObject?>
            ) {
                try {
                    if (response.code() == SUCCESS_CODE_n) {
                        val jsonObject = JSONObject(Gson().toJson(response.body()))
                        if (jsonObject.getString("status")
                                .equals(SUCCESS_CODE, ignoreCase = true)
                        ) {
                            binding.progressBar.visibility = View.GONE
                            storePrefrence.setString(
                                IDENTFIER,
                                jsonObject.getJSONObject("data").getString("identifier")
                            )
                            storePrefrence.setString(TOKEN_LOGIN, "")
                            storePrefrence.setString(
                                "id",
                                jsonObject.getJSONObject("data").getString("id")
                            )
                            val mainIntent = Intent(ctx, DashBoardActivity::class.java)
                            startActivity(mainIntent)
                            finish()
                        } else {
                            binding.progressBar.visibility = View.GONE
                            showToastMessage(Constant.ERRORMSG)
                        }
                    } else if (response.code() == ERROR_CODE) {
                        val jsonObject = JSONObject(response.errorBody()!!.string())
                        if (jsonObject.getString("status")
                                .equals(ERROR_CODE.toString(), ignoreCase = true)
                        ) {
                            binding.progressBar.visibility = View.GONE
                            val errorMsg: String = jsonObject.getString("message")
                            showToastMessage(errorMsg)
                        } else {
                            binding.progressBar.visibility = View.GONE
                            showToastMessage(Constant.ERRORMSG)
                        }
                    }
                } catch (ex: JSONException) {
                    ex.printStackTrace()
                    binding.progressBar.visibility = View.GONE
                    showToastMessage(Constant.ERRORMSG)
                } catch (ex: IOException) {
                    ex.printStackTrace()
                    binding.progressBar.visibility = View.GONE
                    showToastMessage(Constant.ERRORMSG)
                }
            }

            override fun onFailure(call: Call<JsonObject?>, t: Throwable) {
                showToastMessage(Constant.ERRORMSG)
                binding.progressBar.visibility = View.GONE
            }
        })
    }
}